web-bluetooth/gap-characteristics.js

```diff
@@ -0,0 +1,186 @@
+function onButtonClick() {
+  log('Requesting Bluetooth Device...');
+  navigator.bluetooth.requestDevice(
+    {filters: [{ name: '' }], optionalServices: ['generic_access']})
+  .then(device => {
+    log('Connecting to GATT Server...');
+    return device.gatt.connect();
+  })
+  .then(server => {
+    log('Getting Generic Access Profile...');
+    return server.getPrimaryService('generic_access');
+  })
+  .then(service => {
+    log('Getting GAP Characteristics...');
+    return service.getCharacteristics();
+  })
+  .then(characteristics => {
+    characteristics.forEach(characteristic => {
+      switch (characteristic.uuid) {
+
+        case BluetoothUUID.getCharacteristic('gap.appearance'):
+          characteristic.readValue().then(value => {
```

Comment:
I think you need to queue these promises otherwise you are going to get "In Progress" errors.